### PR TITLE
fix(sct_runner.py): use larger disk for long longevity

### PR DESCRIPTION
**Pull request by amoskong**
My 1TB-longevity failed in collecting logs because of ENOSPC on sct
runner. I know it's not a common case, but when it occurs we might lost
the test logs.

Signed-off-by: Amos Kong <amos@scylladb.com>

## PR pre-checks (self review)

- [x] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [x] I didn't leave commented-out/debugging code
- [x] I added the relevant `backport` labels
- [ ] New configuration option are added and documented (in `sdcm/sct_config.py`)
- [ ] I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)
- [ ] All new and existing unit tests passed (CI)
- [ ] I have updated the Readme/doc folder accordingly (if needed)

### CODE REVIEW

**Comment by amoskong** [comment]
BYO: https://jenkins.scylladb.com/job/scylla-staging/job/amos/job/byo-longevity-test/67/

**Review by bentsi — CHANGES_REQUESTED** [review]
please convert the `INSTANCE_ROOT_DISK_SIZE` to property and put the logic inside.
(look at how `instance_type` property is implemented)

**Comment by amoskong** [comment]
> please convert the `INSTANCE_ROOT_DISK_SIZE` to property and put the logic inside.
> (look at how `instance_type` property is implemented)

Fixed.

Retested by byo: https://jenkins.scylladb.com/job/scylla-staging/job/amos/job/byo-longevity-test/71

- set timeout to 5000
- a runner had 55G disk was created

File: sdcm/sct_runner.py [hunk 1]
```diff
@@ -269,6 +276,7 @@ def create_instance(self, test_id: str, test_duration: int, region_az: str):
                 {"Key": "keep_action", "Value": "terminate"},
             ],
             region_az=region_az,
+            test_duration=test_duration,
```
Comment (bentsi) [inline sdcm/sct_runner.py:279]: why did you add the tag? `keep` stores this information

File: sdcm/sct_runner.py [hunk 2]
```diff
@@ -51,6 +50,14 @@ def instance_type(test_duration):
             return "r5.large"
         return "t3.large"  # has 7h 12m CPU burst
 
+    @staticmethod
+    def instance_root_disk_size(test_duration):
+        INSTANCE_ROOT_DISK_SIZE = 35  # GB
```
Comment (bentsi) [inline sdcm/sct_runner.py:55]: we use upper case names for global variables only...
`INSTANCE_ROOT_DISK_SIZE` -> `instance_root_disk_size`

**Review by bentsi — CHANGES_REQUESTED** [review]
please address comments

File: sdcm/sct_runner.py (hunk 1 above)
Comment (amoskong) [reply sdcm/sct_runner.py:279]: I didn't change the tags_list parameter. The test_duration is used inside _create_instance() -> self.instance_root_disk_size().

File: sdcm/sct_runner.py (hunk 2 above)
Comment (amoskong) [reply sdcm/sct_runner.py:55]: instance_root_disk_size is same as the function name, I renamed it to disk_size.

**Review by roydahan — APPROVED** [review]

**Comment by juliayakovlev** [comment]
@roydahan It's not backported to 4.4

**Comment by roydahan** [comment]
Already done.
